src/main/java/com/thoughtworks/gauge/scan/CustomClassInitializerScanner.java

```diff
@@ -0,0 +1,39 @@
+package com.thoughtworks.gauge.scan;
+
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.reflections.Reflections;
+
+import com.thoughtworks.gauge.ClassInitializer;
+import com.thoughtworks.gauge.CustomClassInitializer;
+import com.thoughtworks.gauge.registry.ClassInitializerRegistry;
+
+public class CustomClassInitializerScanner implements IScanner {
+    @Override
+    public void scan(Reflections reflections) {
+        scanForInitializer(reflections);
+    }
+
+    private void scanForInitializer(Reflections reflections) {
+        Set<Class<?>> initializers = reflections.getTypesAnnotatedWith(CustomClassInitializer.class);
+
+        if (initializers.size() > 1) {
+            throw new ScanningException("Only 1 class can be annotated with @CustomClassInitializer");
+        }
+
```

Comment:
This looks good. 
```
18.         } catch (NoSuchElementException e) {
19.             // Class initializer is not defined, use default!
20.             ClassInitializerRegistry.classInitializer(new DefaultClassInitializer());
21.         }
```
Just one thing, instead of setting default class initializer in the catch block, can we check the size of initializers and then assign before using the iterator.
```
        if (initializers.size() < 1) {
            ClassInitializerRegistry.classInitializer(new DefaultClassInitializer());
            return;
        }
```